pkg/tarmak/tarmak.go

```diff
@@ -357,3 +366,134 @@ func (t *Tarmak) CmdKubectl(args []string) error {
+
+func (t *Tarmak) DestroyEnvironment() error {
+	inputDestroy := input.New(os.Stdin, os.Stdout)
+
+	var err error
+
+	destroyClusters := true
+	moveFolder := true
+	removeConfig := true
+
+	if !t.flags.Environment.Destroy.AutoApprove {
+		destroyClusters, err = inputDestroy.AskYesNo(&input.AskYesNo{
+			Default: false,
+			Query:   fmt.Sprintf("Destroy all clusters of %v?", t.Environment().Name()),
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if destroyClusters {
+		t.log.Info("destroying clusters")
+		for _, cluster := range t.Environment().Clusters() {
+			// We first want to destroy the k8s clusters before the hub
+			if cluster.Name() != t.Environment().Hub().Name() {
+				t.cluster = cluster
+				if err = t.DestroyActivecluster(); err != nil {
+					return err
+				}
+			}
+		}
+
+		// After destroying all other clusters, we can destroy the hub
+		t.cluster = t.Environment().Hub()
+		if err = t.DestroyActivecluster(); err != nil {
+			return err
+		}
+	} else {
+		for _, cluster := range t.Environment().Clusters() {
+			hosts, err := cluster.ListHosts()
+			if err != nil {
+				return err
+			}
+			if len(hosts) > 0 {
+				return fmt.Errorf("can't proceed with destroying this environment, because it still has hosts running")
+			}
+		}
+	}
+
+	if !t.flags.Environment.Destroy.AutoApprove {
+		moveFolder, err = inputDestroy.AskYesNo(&input.AskYesNo{
+			Default: false,
+			Query:   "Move environment folder (SSH key and vault_root_token) to .archive?",
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if moveFolder {
+		t.log.Info("moving environment folder to .archive")
+
+		archivePath := filepath.Join(t.ConfigPath(), ".archive")
+		environmentArchivePath := filepath.Join(archivePath, t.Environment().Name())
+
+		if _, err = os.Stat(archivePath); err != nil {
+			if os.IsNotExist(err) {
+				if err = os.MkdirAll(archivePath, os.ModePerm); err != nil {
+					return err
+				}
+			} else {
+				return fmt.Errorf("uncatched error: %v", err)
+			}
+		}
+
+		if _, err = os.Stat(environmentArchivePath); err != nil {
+			if os.IsNotExist(err) {
+				if err = os.Rename(t.Environment().ConfigPath(), environmentArchivePath); err != nil {
+					return err
+				}
+			} else {
+				return fmt.Errorf("uncatched error: %v", err)
+			}
+		} else {
+			return fmt.Errorf("already archived %v", t.Environment().Name())
+		}
+	}
+
+	if !t.flags.Environment.Destroy.AutoApprove {
+		removeConfig, err = inputDestroy.AskYesNo(&input.AskYesNo{
+			Default: false,
+			Query:   "Remove environment from tarmak.yaml?",
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if removeConfig {
+		t.log.Infof("removing environment %v from tarmak.yaml", t.Environment().Name())
+
+		if err = t.config.RemoveEnvironment(t.Environment().Name()); err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func (t *Tarmak) DestroyActivecluster() error {
+	var args []string
+
+	t.log.Infof("Destroying cluster %v", t.Cluster().Name())
+	destroyCmd := t.NewCmdTerraform(args)
```

Comment:
We are not passing signals properly here. The destroy was not respecting my interrupts

Reply:
Going to use this solution https://github.com/jetstack/tarmak/pull/575#discussion_r225991922